pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -407,6 +407,9 @@ func (e *EndpointController) syncService(key string) error {
 	}
 	service, err := e.serviceLister.Services(namespace).Get(name)
 	if err != nil {
+		if !errors.IsNotFound(err) {
```

Comment:
actually, the change short-circuits the following b/c for now an error from lister is always a not-found error. so i see your point as deleting L413-423

Reply:
If it is not-found error, we need L413-423, see https://github.com/kubernetes/kubernetes/pull/58033. If it always a not-found error, close it now. Thanks!